src/lib/qdl.js

```diff
@@ -0,0 +1,51 @@
+const execa = require('execa');
+const utilities = require('../lib/utilities');
+const path = require('path');
+const fs = require('fs-extra');
+const os = require('os');
+const { update } = require('lodash');
+
+const TACHYON_STORAGE_TYPE = 'ufs';
+
+async function getExecutable() {
+	const archType = utilities.getArchType();
+	const archName = utilities.getOs();
+	const qdlExec = path.join(__dirname, `../../assets/qdl/${archName}/${archType}/qdl`);
+	await fs.ensureFile(qdlExec);
+	await fs.chmod(qdlExec, 0o755);
+	return qdlExec;
+}
+
+/**
+ */
+async function run({ files, updateFolder, zip, verbose, ui }) {
+	const qdl = await getExecutable();
+
+	if (zip) {
```

Comment:
Make sure that the caller doesn't pass a slash in front of all the files instead of removing it here.